### DISCUSSION TITLE
Adding Auxiliary Task (Additional Loss)

### DISCUSSION THREAD

**Discussion by ErcBunny** [Toni-SM/skrl#171] [Q&A] [answered]
How should a user implement auxiliary tasks that uses additional self-supervision to optimize policy networks during training? For example, one might want to use image reconstruction loss to provide additional supervision for vision encoders, while using PPO loss at the same time. Is there a recommended way of realizing this?

Thanks for your thoughts and comments.

**Comment by Toni-SM** [comment, marked as answer]
Hi @ErcBunny 

I think that for this will be necessary to modify the desired agent implementation
For example, for the PPO agent, compute the additional loss and add it to:
https://github.com/Toni-SM/skrl/blob/636936f3ac49c6d2260bd130d72b789ca6dfe42b/skrl/agents/torch/ppo/ppo.py#L429
